avocado/utils/configure_network.py

```diff
@@ -10,193 +10,323 @@
-            raise NWException("%s interface not available" % interface)
-        with open(conf_file, "w") as network_conf:
-            if interface_type is None:
-                interface_type = 'Ethernet'
-            network_conf.write("TYPE=%s \n" % interface_type)
-            network_conf.write("BOOTPROTO=none \n")
-            network_conf.write("NAME=%s \n" % interface)
-            network_conf.write("DEVICE=%s \n" % interface)
-            network_conf.write("ONBOOT=yes \n")
-            network_conf.write("IPADDR=%s \n" % ipaddr)
-            network_conf.write("NETMASK=%s \n" % netmask)
-            network_conf.write("IPV6INIT=yes \n")
-            network_conf.write("IPV6_AUTOCONF=yes \n")
-            network_conf.write("IPV6_DEFROUTE=yes")
-
-        cmd = "ifup %s" % interface
+
+    def __init__(self, if_name, if_type='Ethernet', remote_session=None):  # pylint: disable=W0231
+        self.name = if_name
+        self.if_type = if_type
+        self.remote_session = remote_session
+
+    def _get_interface_details(self, version=4):
+        output = ''
+        cmd = "ip -{} -j address show {}".format(version, self.name)
+        output = _run_command(cmd, self.remote_session)
         try:
-            process.system(cmd, ignore_status=False, sudo=True)
-            return True
-        except process.CmdError as ex:
-            raise NWException("ifup fails: %s" % ex)
+            return json.loads(output)
+        except (KeyError, json.JSONDecodeError):
+            msg = "Unable to get IP address on interface {}".format(self.name)
+            log.error(msg)
+            raise NWException(msg)
 
-    if distro.detect().name == 'SuSE':
-        conf_file = "/etc/sysconfig/network/ifcfg-%s" % interface
-        if os.path.exists(conf_file):
-            shutil.move(conf_file, conf_file+".backup")
+    def _move_file_to_backup(self, filename, ignore_missing=True):
+        destination = "{}.backup-{}".format(filename, time.time())
+        if os.path.exists(filename):
+            shutil.move(filename, destination)
         else:
-            raise NWException("%s interface not available" % interface)
-        with open(conf_file, "w") as network_conf:
-            network_conf.write("IPADDR=%s \n" % ipaddr)
-            network_conf.write("NETMASK='%s' \n" % netmask)
-            network_conf.write("IPV6INIT=yes \n")
-            network_conf.write("IPV6_AUTOCONF=yes \n")
-            network_conf.write("IPV6_DEFROUTE=yes")
-
-        cmd = "ifup %s" % interface
+            if not ignore_missing:
+                raise NWException("%s interface not available" % self.name)
+
+    def _write_to_file(self, filename, values):
+        self._move_file_to_backup(filename)
+
+        with open(filename, 'w+') as fp:
+            for key, value in values:
+                fp.write("{}={}\n".format(key, value))
+
+    def add_hwaddr(self, hwaddr):
+        """
+        Utility which add mac address to a interface return Status
```

Comment:
@apahim maybe you are looking to an outdated version. Since that this is a draft PR, I'm still doing pushes here. For instance, this was already changed.

Reply:
Interactive rebase before pushing ftw.